janmatyoo/ev-employee-onboarding-exercise
# PR for EVOBE-126,  EVOBE-127, EVOBE-128

### Ticket
<https://phunts.atlassian.net/browse/EVOBE-126>
<https://phunts.atlassian.net/browse/EVOBE-127>
<https://phunts.atlassian.net/browse/EVOBE-128>

### Changes
- Setup the folder structure
- Added automatic reloading using nodemon
- Added default logging using morgan
- Added Error handling

### Steps to Test
1. Pull branch
2. Go to the folder `backend`
3. Run the cli command `npm install` to install all the dependencies
4. Run the cli command `npm start` to start the app
5.  For initial testing of GET API for list all employees, Go to the url `localhost:80/employees` and use the `GET` method on postman, it should display the following json response:
```javascript
{
    "message": "List all employees"
}
```
6. For initial testing of GET API for get specific employee, Go to the url `localhost:80/employees/<chosen-id>`and use the `GET` method on postman, it should display the following json response:
```javascript
{
    "message": "List specific employee"
}
```
7.  For initial testing of POST API for adding new employee, Go to the url `localhost:80/employees` and use the `POST` method and empty body on postman, it should display the following json response:
```javascript
{
    "message": "Added employee"
}
```
8. For initial testing of PUT API for updating selected employee, Go to the url `localhost:80/employees/<chosen-id>` and use the `PUT` method and empty body on postman, it should display the following json response:
```javascript
{
    "message": "Updated employee"
}
```
9. For initial testing of DELETE API for deleting selected employee, Go to the url `localhost:80/employees/<chosen-id>` and use the `DELETE` method on postman, it should display the following json response:
```javascript
{
    "message": "Deleted employee"
}
```

### Resources
- N/A
